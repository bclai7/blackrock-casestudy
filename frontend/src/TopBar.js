import Button from 'react-bootstrap/Button';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

function TopBar() {

  return (
    <div className="TopBar sectionHeader" data-testid="topBar">
        <Row>
            <Col xs={{ span: 4 }}>
                AI Labs | <b>Full Stack Case Study</b>
            </Col>
            <Col xs={{ span: 1, offset: 7 }}>
                <Button className='runButton' type="submit" variant="primary" data-testid="runButton">
                    Run
                </Button>
            </Col>
            
        </Row>

    </div>
  );
}

export default TopBar;
