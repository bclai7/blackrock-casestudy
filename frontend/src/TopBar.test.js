import { render, screen, cleanup } from '@testing-library/react';
import TopBar from './TopBar';
import renderer from 'react-test-renderer';

test('renders TopBar component', () => {
  render(<TopBar />);
  const topBarElement = screen.getByTestId('topBar');
  expect(topBarElement).toBeInTheDocument();
  expect(topBarElement).toHaveTextContent('AI Labs | Full Stack Case Study');
  const runButton = screen.getByTestId('runButton');
  expect(runButton).toBeInTheDocument();
  expect(runButton).toHaveTextContent('Run');
});

test('topBar snapshot', () => {
    const snap = renderer.create(<TopBar />).toJSON();
    expect(snap).toMatchSnapshot();
});
