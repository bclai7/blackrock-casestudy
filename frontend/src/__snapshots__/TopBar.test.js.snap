// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`topBar snapshot 1`] = `
<div
  className="TopBar sectionHeader"
  data-testid="topBar"
>
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      AI Labs | 
      <b>
        Full Stack Case Study
      </b>
    </div>
    <div
      className="col-1 offset-7"
    >
      <button
        className="runButton btn btn-primary"
        data-testid="runButton"
        disabled={false}
        type="submit"
      >
        Run
      </button>
    </div>
  </div>
</div>
`;
